### ISSUE TITLE
unable to orderBy on many-to-many relation

### ISSUE THREAD

**Issue by vesper8** [vuex-orm/vuex-orm#503]
I have a `User` model who has a many-to-many relationship with a `Skill` model. I use a pivot model called `UserSkill`

Everything works fine, inserts, retrieving using `withAll()` or `with('skills)`

One thing I can't seem to get to work is retrieving the skills ordered by `'name'`

I have tried this

```
const users = User.query().with('skills', (query) => {
  query.orderBy('name')
}).get()

console.log(users[0].skills);
```

Same for a single user:
```
      this.user = User.query()
        .with('skills', (query) => {
          query.orderBy('name')
        })
        .where('id', 1)
        .first();

        console.log(user.skills);
```

It returns the skills but they are not ordered

I also tried doing the reverse

```
const skills = Skill.query().whereHas('users', (query) => {
              query.where('userId', 1;
            }).get()
```

This however doesn't work at all and returns an empty result set. I must be doing something wrong there and this query is not taking into account the many-to-many nature of the relationship

I'm more interested in getting the first approach to work... is there a way?

**Comment by kiaking** [comment]
Oh, this should work. Need to create some test case for this.

**Comment by cuebit** [comment]
Resolved by #586
